File: JAVA/41. First Missing Positive.java
class Solution {
    public int firstMissingPositive(int[] n) {
        Arrays.sort(n);
        int i=1;
        for(int x:n){
            if(i==x)
            i++;
            else if(x>i)
            break;
        }
        return i;    
    }
}

File: C++/41. First Missing Positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& n) {
        set<int> a(n.begin(), n.end());
        int i = 1;
        while (true) {
            if (a.find(i) == a.end()) return i;
            i++;
        }
        return *max_element(n.begin(),n.end())+1;
    }
};
